frontend/src/ts/controllers/sound-controller.ts

```diff
@@ -341,6 +347,71 @@ function initAudioContext(): void {
   }
 }
 
+const scales: Record<string, string[]> = {
+  pentatonic: ["C", "D", "E", "G", "A"],
+};
+
+const initialScale: ScaleData = {
+  //Metadata for the scale
+  position: 0,
+  octave: 4,
+  direction: 1,
+};
+
+function createPreviewScale(num: number): () => void {
+  // We use a JavaScript closure to create a preview function that can be called multiple times and progress through the scale
+  const scale: ScaleData = {
+    position: 0,
+    octave: 4,
+    direction: 1,
+  };
+  return function previewScale() {
+    if (num == 12) {
+      playPentatonic(scale);
+    }
+  };
+}
+export const previewPentatonic: () => void = createPreviewScale(12);
+
+export function playPentatonic(scaleMeta: ScaleData): void {
+  if (audioCtx === undefined) {
+    initAudioContext();
+  }
+  if (!audioCtx) return;
+
+  const randNote: number = Math.floor(
+    Math.random() * scales["pentatonic"].length
+  );
```

Comment:
We have utilities for random values:

```suggestion
    const randomNote = randomIntFromRange(0, scales["pentatonic"].length)
```